onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -1479,6 +1479,45 @@ Status CastOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const N
 
 #pragma endregion
 
+#pragma region op_depthtospace
+class DepthToSpaceOpBuilder : public BaseOpBuilder {
+ private:
+  Status AddToModelBuilderImpl(ModelBuilder& model_builder, const NodeUnit& node_unit) const override;
+};
+
+Status DepthToSpaceOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const NodeUnit& node_unit) const {
+  auto& shaper(model_builder.GetShaper());
+  const auto& operand_indices(model_builder.GetOperandIndices());
+  const auto& operand_types(model_builder.GetOperandTypes());
+  const auto android_feature_level = model_builder.GetNNAPIFeatureLevel();
+  NodeAttrHelper helper(node_unit);
+
+  const auto& input = node_unit.Inputs()[0].node_arg.Name();
+  const auto& output = node_unit.Outputs()[0].node_arg.Name();
+
+  bool use_nchw = model_builder.UseNCHW();
+  ORT_RETURN_IF_ERROR(IsOpInRequiredLayout(use_nchw, node_unit));
+
+  const auto blocksize = helper.Get("blocksize", 2);
+
```

Comment:
Is 2 a default value? The spec says the blocksize attribute is required but doesn't specify a default.

Reply:
No 2 is not a default value for blocksize. didn't find anywhere specifying the default value. 

But as blocksize >=1 and looked up all the operator tests used blocksize = 2 so I just put 2 here. Maybe any other better ways to do this?

Reply:
I would copy the implementation from NodeAttrHelper instead of providing a fake default. You know a value exists as the onnx spec requires it, so the onnx checker would have validated that when the node was loaded.

`node_unit.GetNode().GetAttributes().at("blocksize").i();`

There's also a chance this code is not doing what you think it is. NodeAttrHelper.Get has both int32_t and int64_t versions. You aren't explicitly specifying a type anywhere here so do you know which one is matched?

In this case it doesn't matter, as the attribute can only be int64_t (the int32_t Get is mainly providing a conversion). But if that wasn't the case, you could have matched the wrong one (i.e. a non-existent attribute due to having the wrong type), and because you provided a default that would have happily been returned.

This is also a flaw in the tests. If they all use a blocksize of 2 you wouldn't notice this sort of problem.

Reply:
thanks for correcting me of this.  didn't think deeply about this before... indeed the node attribute helper can only use a default value which is a value that actually exists...

updated it as required.